### ISSUE TITLE
Suggestion: a second button for the hotspot and object interactions that stays in the room editor

### ISSUE THREAD

**Issue by selmiak** [adventuregamestudio/ags#1451]
when handing the different interactions with a hotspot or an object AGS jumps from the room editor to the script of the room and adds a few lines of code for the function. As I tend to add all possible interactions for one hotspot or object all at once this leads to a lot of back and forth and is just exhausting, can we please have a button that adds these lines to the script and stays in the room editor. Ot does the editor need to be in the script to add these lines? I wouldn't mind when the room script open on the button press, but in the background, not as the focus tab. It could look something like this:

![ags_add](https://user-images.githubusercontent.com/22618469/141870187-b3ed0c52-5355-464c-9c01-6b04f532417a.png)

**Comment by ivan-mogilko** [comment]
Alternative approach would perhaps be to handle input combinations, e.g. Shift + Click on the same button; although I do not know how this button is handled at all in this control, and whether such events are possible.

**Comment by selmiak** [comment]
Ctrl+Click could also work, if that is easier to put in please add that. The disadvantage for this hotkey version of implementation is that the function of the editor is not obviously visible, you have to _know_ this functionality. That is not intuitive for the user. Can we have tooltips in AGS by now? That would be really useful to tell the user that this is also possible in case of the hotkey combination implementation. 
Seeing this disadvantage I'd prefer the second button implementation, but appreciate every implementation, whatever is easier to put into the editor. I like that ... on the button is a bit like the .. directory, that is somewhere else, and the . directory is stay in that directory, in that case, stay in that editor window.

The good thing is, you don't need any additional feedback that something happened in the script as you can see the functionality being added to that action-field of the hotspot/object. 

Also let me add, on top of **Hotspots** and **Objects**, **Inventory Items** need that same functionality of the editor, and **GUI-buttons** and maybe some more game objects I don't think about right now too.

**Comment by ivan-mogilko** [comment]
> Also let me add, on top of Hotspots and Objects, Inventory Items need that same functionality of the editor, and GUI-buttons and maybe some more game objects I don't think about right now too.

Obviously, this is not going to be added to only one object type, the property grid is one control that is used for everything, so these new controls, if added, will work everywhere same way.

**Comment by selmiak** [comment]
good to know. I thought that maybe they have to be implemented per object type and was trying to list these.

**Comment by ericoporto** [comment]
The behavior of going to the script is done here: https://github.com/adventuregamestudio/ags/blob/master/Editor/AGS.Types/PropertyGridExtras/ScriptFunctionUIEditor.cs#L75

I read [this guide here](https://docs.microsoft.com/en-us/previous-versions/dotnet/articles/aa302326(v=msdn.10)?redirectedfrom=MSDN) but could not figure how to add a second button. I found a second type of interaction called [DesignerVerb](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.design.designerverb?redirectedfrom=MSDN&view=netframework-4.6.2), which adds like a link in the property descriptor that can do things. What I could think of is adding a preference to disable automatic jumping to script editor and let the design verb link handle jumping to script editor.
